File: lcr/165/main.go
package main

import "strconv"

func crackNumber(ciphertext int) int {
	src := strconv.Itoa(ciphertext)
	a, b := 1, 1

	for i := 2; i <= len(src); i++ {
		tmp := src[i-2 : i]
		var c int
		if tmp >= "10" && tmp <= "25" {
			c = a + b
		} else {
			c = b
		}
		a = b
		b = c
	}
	return b
}

File: lcr/165/main.py
class Solution:
    def crackNumber(self, ciphertext: int) -> int:
        s = str(ciphertext)

        a, b = 1, 1
        for i in range(2, len(s)+1):
            tmp = s[i-2:i]
            a, b = (a+b, a) if "10" <= tmp <= "25" else (a, a)

        return a
